x/participationrewards/keeper/hooks.go

```diff
@@ -146,6 +149,105 @@ func (k *Keeper) AfterZoneCreated(ctx sdk.Context, zone *icstypes.Zone) error {
 		Data: localDenomBytes,
 	})
 
+	// Create LiquidAllowedDenomProtocolData for the host zone
+	hostZoneDenom := types.LiquidAllowedDenomProtocolData{
+		ChainID:               zone.ChainId,
+		RegisteredZoneChainID: zone.ChainId,
+		IbcDenom:              zone.BaseDenom,
+		QAssetDenom:           zone.LocalDenom,
+	}
+	if err := hostZoneDenom.ValidateBasic(); err != nil {
+		return err
+	}
+	hostZoneDenomBytes, err := json.Marshal(hostZoneDenom)
+	if err != nil {
+		return err
+	}
+	k.SetProtocolData(ctx, hostZoneDenom.GenerateKey(), &types.ProtocolData{
+		Type: types.ProtocolDataType_name[int32(types.ProtocolDataTypeLiquidToken)],
+		Data: hostZoneDenomBytes,
+	})
+
+	// Fetch OsmosisParamsProtocolData and create LiquidAllowedDenomProtocolData for Osmosis
+	k.IteratePrefixedProtocolDatas(ctx, types.GetPrefixProtocolDataKey(types.ProtocolDataTypeOsmosisParams), func(index int64, _ []byte, data types.ProtocolData) (stop bool) {
+		osmosisParams, err := types.UnmarshalProtocolData(types.ProtocolDataTypeOsmosisParams, data.Data)
+		if err != nil {
+			return false
+		}
+		osmosisParamsData, ok := osmosisParams.(*types.OsmosisParamsProtocolData)
+		if !ok {
+			return false
+		}
+		osmosisConnectionData, found := k.GetProtocolData(ctx, types.ProtocolDataTypeConnection, osmosisParamsData.ChainID)
+		if !found {
+			return false
+		}
+		connectionData, err := types.UnmarshalProtocolData(types.ProtocolDataTypeConnection, osmosisConnectionData.Data)
+		if err != nil {
+			return false
+		}
+		osmosisChannel := connectionData.(*types.ConnectionProtocolData).TransferChannel
+
+		osmosisDenom := types.LiquidAllowedDenomProtocolData{
+			ChainID:               osmosisParamsData.ChainID,
+			RegisteredZoneChainID: zone.ChainId,
+			IbcDenom:              utils.DeriveIbcDenom(transfertypes.PortID, osmosisChannel, "", "", zone.LocalDenom),
```

Comment:
this will result in the incorrect value; the value has to be computed as if from the counterpary side:

```
channel, found := k.IBCKeeper.ChannelKeeper.GetChannel(ctx, "transfer", osmosisChannel)
...
utils.DeriveIbcDenom("transfer", channel.Counterparty.ChannelId, "transfer", osmosisChannel, zone.LocalDenom)
```